tests/cobs_test.cc

```diff
@@ -0,0 +1,145 @@
+#include "../inc/cobs.h"
+
+/*
+Throw exceptions so we can catch them with gdb (catch catch).
+*/
+void throw_assert(bool condition, const char * const remark) {
+  if (condition == false) throw remark;
+}
+
+/*
+Fill a byte buffer with a given value.
+*/
+void fill(uint8_t * const b, size_t l, uint8_t v) {
+  for (size_t i = 0; i < l; i++) {
+    b[i] = v;
+  }
+}
+
+/*
+Check that a byte buffer contains only a given value.
+*/
+void cmpb(uint8_t * const b, size_t l, uint8_t v) {
+  for (size_t i = 0; i < l; i++) {
+    throw_assert(b[i] == v, "");
+  }
+}
+
+int main() {
+  const uint8_t X = 0xdd; // non-zero payload
+  const uint8_t F = 0xcc; // fill
+
+  // Encode the empty packet.
+  {
+    uint8_t b1[1000]; fill(b1, sizeof(b1), X);
```

Comment:
This is a by product of not being able to declare an array filled with a given value. 

If you would accept

```cpp
list<uint8_t> b = new list<uint8_t>(1000).fill(X);
```

Then the current way of writing it should be fine too.